backend/app/api/localdatasets.py

```diff
@@ -0,0 +1,43 @@
+import logging
+import csv
+from typing import List
+
+DATA_PATH='app/data/'
+output_max=10
+def get_unesco_sites(gpe: str):
```

Comment:
Probably entire code could be simplified.

```py
from typing import Any

def get_data_from_csv(filename: str, gpe: str, row_func: Callable[[Any], str]):
     with open(f"{DATA_PATH}{filename}.csv", 'r') as csvfile:
        reader = csv.DictReader(csvfile, delimiter=';')
        data_list: list[str] = []
        for row in reader:
            res = row_func(row, gpe)
            if res is not None:
               data_list.append(res)
            if len(data_list) >= output_max:
               break
        return data_list
```

And then you'd have for example:

```py
def _get_historical_places(row, gpe: str):
   if row.get('County', "").lower() == gpe.lower():
         return row['Resource_Name']
   return None

def get_historical_places(gpe: str):
   return get_data_from_csv('US_Historic_Places', gpe, _get_historical_places)
```
Or could use lambda function